diff -r 90440909355b cmake/Modules/FindOptiX.cmake
--- a/cmake/Modules/FindOptiX.cmake	Thu Aug 16 14:39:27 2018 +0800
+++ b/cmake/Modules/FindOptiX.cmake	Thu Aug 16 15:48:15 2018 +0000
@@ -35,6 +35,12 @@
   set(bit_dest "")
 endif()
 
+
+if (DEFINED ENV{OptiX_INSTALL_DIR})
+    set(OptiX_INSTALL_DIR $ENV{OptiX_INSTALL_DIR})
+endif()
+
+
 macro(OPTIX_find_api_library name version)
   find_library(${name}_LIBRARY
     NAMES ${name}.${version} ${name}
