diff -r 659f02f409ac cmake/Modules/EnvCompilationFlags.cmake
--- a/cmake/Modules/EnvCompilationFlags.cmake	Sun Dec 17 16:54:21 2017 +0800
+++ b/cmake/Modules/EnvCompilationFlags.cmake	Thu Jan 11 18:16:19 2018 +0000
@@ -25,7 +25,7 @@
       #set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
       #set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -std=c++0x")
      # set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -std=c++11")   #needed for numpyserver- on Linux ?
-     set(CMAKE_CXX_STANDARD 14)
+     set(CMAKE_CXX_STANDARD 11)
      set(CMAKE_CXX_STANDARD_REQUIRED on)
 
   endif ()
diff -r 659f02f409ac opticks.bash
--- a/opticks.bash	Sun Dec 17 16:54:21 2017 +0800
+++ b/opticks.bash	Thu Jan 11 18:16:19 2018 +0000
@@ -375,7 +375,7 @@
 } 
 
 
-opticks-compute-capability(){ echo ${OPTICKS_COMPUTE_CAPABILITY:$($FUNCNAME-)} ; }
+opticks-compute-capability(){ echo ${OPTICKS_COMPUTE_CAPABILITY:-$($FUNCNAME-)} ; }
 opticks-compute-capability-()
 {
     local t=$NODE_TAG
